CONTRIBUTING.md

```diff
@@ -0,0 +1,296 @@
+
+#### Squashing Commits
+
+1. First, find out how many commits there are on your branch:
+   ```
+   git log --oneline master..your-branch-name | wc -l
+   ```
+2. Then, use an interactive rebase to squash all but the first commit unless
+   you have a good reason to keep more than one:
+   ```
+   git rebase -i HEAD~#
+   ```
+   where `#` is the number of commits on your branch. Mark the first commit as
+   `r` (for reword) so you can update the message and the rest as `s` (for
+   squash).
+
+In terms of the commit message, we recommend the following:
+* The first line should summarize the impact of the commit and should be less
+  than 72 characters. Note that this will likely go into the Changelog as well.
+  * This should also be in the present tense / imperative mood (e.g. "Add
+    feature" vs "Added feature" and "Change element" vs "Changes element").
+* Leave the second line blank
+* The third line should reference the issue this PR resolves, either
+  `issue #1234` if you're just mentioning the issue or `closes #1234` if
+  you're closing an issue.
+  * If you don't have an issue to reference, make sure you don't need to open
+    one before opening the PR.
+* Use bullet points on the following lines to elaborate on your brief
+  description.
+
+Example:
+```
+Fix the very important thing that was broken
+
+closes #1234
+- change the first file
+- change the second file
+- take this other thing into account
+- modify this other thing
+```
+
+#### Merge Conflicts
+After you've squashed commits and crafted the perfect commit message, you need
+to resolve any merge conflicts with the upstream branch. First, make sure that
+you have latest version of `master` (or your upstream branch) checked out and
+then rebase your squashed branch on top of it:
+```
+git checkout master
+git pull upstream master
+git checkout 1234_your_branch_name
+git rebase master
+```
+
+Now just push your branch up to GitHub. If you've already pushed it prior to
+rebasing, you'll need to force push it (using the `-f` flag, **CAREFULLY**),
+otherwise you should be able to just push it normally:
+```
+git push origin 1234_your_branch_name
+```
+
+#### Testing
+Before finally submitting, make sure that your branches passes all of our
+tests and style checks:
+```
+rake
+rubocop -D
+```
+
+### Submitting your PR
+At this point you have a branch that is based off of the latest version of
+your upstream branch with a clean history and useful commit message. If you go
+to your fork's page on GitHub, there should be a prompt to submit a PR; if not
+go [here](https://github.com/YaleSTC/reservations/compare), select
+`YaleSTC/reservations` as the base fork and the relevant base branch and your
+fork as the head fork with your branch as the compare brach. Click "Create
+Pull Request" and add a description which references the original issue (e.g.
+`Resolves #1234`), describes the changes you made and any new tools or
+libraries you added, as well as any breaking changes.
```

Comment:
I think anywhere we have a list of steps (like this section) we should have bullet points